src/test/java/org/broadinstitute/hellbender/utils/nio/NioTests.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.gcloud.AuthCredentials;
+import com.google.gcloud.RetryParams;
+import com.google.gcloud.storage.StorageOptions;
+import com.google.gcloud.storage.contrib.nio.CloudStorageConfiguration;
+import com.google.gcloud.storage.contrib.nio.CloudStorageFileSystem;
+
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Test GCS access via the NIO APIs.
+ */
+public final class NioTests extends BaseTest {
+
+  @Test
+  public void testGcsEnabled() {
+    FileSystem fs = FileSystems.getFileSystem(URI.create("gs://domain-registry-alpha"));
+  }
+
+  @Test
+  public void openPublicFile() throws IOException {
+    InputStream inputStream = Files.newInputStream(Paths.get(URI.create("gs://pgp-harvard-data-public/hu011C57/GS000018120-DID/GS000015172-ASM/manifest.all.sig")));
+    int firstByte = inputStream.read();
+  }
+
+  @Test
+  public void OpenPrivateFile() throws IOException {
+    // this file, potentially unlike the others in the set, is not marked as "Public link".
+    final String privateFile = getGCPTestInputPath() + "org/broadinstitute/hellbender/utils/private_file.txt";
+    final String BUCKET = privateFile.split("/")[2];
+    FileSystem fs = getAuthenticatedGcs(BUCKET);
+    Path path = fs.getPath(privateFile);
+    int firstByte = Files.newInputStream(path).read();
+  }
+
+  private FileSystem getAuthenticatedGcs(String bucket) throws IOException {
```

Comment:
It looks like it might be possible to call:
` CloudStorageFileSystemProvider.setGCloudOptions(authenticatedStorageOptions);`

I'm not sure if FileSystems are created on demand or if they are initialized eagerly, and setting the provider options during execution may be too late.